js/stats.js

```diff
@@ -1,80 +1,75 @@
-// };
-//
-// stats.numOfAuthors = function() {
-//   var authorCount = stats.uniqueAuthors.length;
-//   $('main').append('<p>Number of Articles: ' + authorCount + '</p>');
-// };
-//
-// stats.numberOfAuthorWords = function() {
-//   stats.uniqueAuthors.forEach(function(element, index, array) {
-//     var countTemp = 0;
-//     var getAuthorWords = function(art) {
-//       if (art.author === element) {
-//         countTemp += stats.wordCount(art.markdown).length;
-//       };
-//     };
-//     stats.allArticles.forEach(getAuthorWords);
-//     stats.eachAuthorWords.push(countTemp);
-//   });
-//
-//   stats.uniqueAuthors.forEach(function(element, index, array) {
-//     $('main').append('<p>Number of Words ' + element + ' has written: </p>');
-//   });
-// };
-//
-// $(function() {
-//   $.getJSON('/util/articles.json', stats.getAllArticles)
-//   .done(stats.numOfArticles);
-//   .done(stats.searchAuthors);
-//   .done(stats.numOfAuthors);
-//   .done(stats.totalWordCount);
-//   .done(stats.allWordLengths);
-//   .done(stats.numberOfAuthorWords);
-// });
-//
-// stats.wordCount = function(str) {
-//   return str.replace(/[#,\n]/g,'').match(/\b\w+/g);
-// };
+var stats = {};
+stats.words = [];
+var uniqueAuthor;
+var totalWords;
+
+$.getJSON('/util/articles.json', function(data){
+  stats.data = data;
+})
+  .done(function(){
+    stats.totalArticles();
+    stats.totalAuthors();
+    stats.totalWords();
+    stats.avgOverallWordLength();
+    stats.wordsByAuthor();
+  });
+
+stats.avgOverallWordLength = function() {
+  var totalWordLength = stats.words.map(function(item) {
+    return item.split('').length;
+  });
+  var totalWordLengthNumber = totalWordLength.reduce(stats.count);
+  var averageWordLength = Math.round(totalWordLengthNumber/totalWords);
+  $('#avgOverall').html('<p>Average word length across all posts: ' + averageWordLength + ' letters</p>');
+};
+
+stats.count = function(a,b) {
+  return a + b;
+};
+
+stats.pluck = function(property, collection) {
+  return collection.map(function(e){
+    return e[property];
+  });
+};
+
+stats.totalArticles = function() {
+  $('#totalArticles').html('<p>Total articles: ' + stats.data.length + '</p>');
+};
+
+stats.totalAuthors = function() {
+  stats.author = [];
+  stats.author = stats.pluck('author', stats.data);
+  uniqueAuthor = $.unique(stats.author);
+  $('#totalAuthors').html('<p>Total authors: ' + uniqueAuthor.length + '</p>');
+};
+
+stats.totalWords = function() {
+  stats.words = stats.pluck('markdown', stats.data);
+  var wordCount = stats.words.map(function(item) {
+    return item.split(' ').length;
+  });
+  totalWords = wordCount.reduce(stats.count);
+  $('#totalWords').html('<p>Total words: ' + totalWords + '</p>');
+};
+
+stats.wordsByAuthor = function() {
+  for(var i=0; i<uniqueAuthor.length; i+=1) {
```

Comment:
Technical requirements for this assignment state: `Do not use the for keyword! Use the functional programming methods.` Can you find a way to compose this method without using the `for` keyword?